tests/test_bytes.py

```diff
@@ -21,7 +30,66 @@ def test_repr():
     [bytes([i]) for i in range(256)],
 )
 def test_uno_byte_bytes_repr(b: bytes) -> None:
+    """Test the repr of Bytes and bytes for single byte values."""
     rust_bytes = Bytes(b)
     rust_bytes_str = repr(rust_bytes)
     rust_bytes_str_eval = eval(rust_bytes_str)  # noqa: S307
     assert rust_bytes_str_eval == rust_bytes == b
+
+
+class TestBytesRemovePrefixSuffix:
+    """Test the remove_prefix and remove_suffix methods."""
+
+    def test_remove_prefix(self) -> None:
+        """Test that remove_prefix works as expected."""
+        rust_bytes = Bytes(b"asdf")
+        assert rust_bytes.removeprefix(b"as") == Bytes(b"df")
+        assert rust_bytes.removeprefix(b"asdf") == Bytes(b"")
+
+    def test_remove_suffix(self) -> None:
+        """Test that remove_suffix works as expected."""
+        rust_bytes = Bytes(b"asdf")
+        assert rust_bytes.removesuffix(b"df") == Bytes(b"as")
+        assert rust_bytes.removesuffix(b"asdf") == Bytes(b"")
+
+
+class TestBytesSlice:
+    """Test suite for Bytes slicing."""
+
+    def test_zero_step_value_err(self) -> None:
+        """Test that slicing with step=0 raises ValueError."""
+        rs_bytes = Bytes(b"abcdefg")
+        py_bytes = b"abcdefg"
+        with pytest.raises(ValueError, match="slice step cannot be zero"):
+            _py_new = py_bytes[0:4:0]
+
+        with pytest.raises(ValueError, match="slice step cannot be zero"):
+            _rs_bytes = rs_bytes[0:4:0]
+
+    @pytest.mark.parametrize(
+        "py_bytes",
+        [b"abcdefg", b"", ALL_BYTES],
+    )
+    def test_slice_o_bytes(self, py_bytes: bytes) -> None:
+        """Run slicing on both bytes and Bytes and assert they are equal."""
+        rs_bytes = Bytes(py_bytes)
+        for start, stop, step, _sliced in _bytes_slices(py_bytes):
+            new_py = py_bytes[start:stop:step]
+            new_rs = rs_bytes[start:stop:step]
+            assert new_rs == new_py
+
+
+def _bytes_slices(
```

Comment:
If this is only used in `TestBytesSlice` and isn't intended to be used by other tests in the future, can you indent this as part of the class and decorate it with `@staticmethod`?